RevenueCatUI/CustomerCenter/ViewModels/BaseManageSubscriptionViewModel.swift

```diff
@@ -108,9 +109,12 @@ class BaseManageSubscriptionViewModel: ObservableObject {
                 }
 
         case let .promotionalOffer(promotionalOffer) where purchaseInformation?.store == .appStore:
-            if promotionalOffer.eligible {
+            if promotionalOffer.eligible, let productIdentifier = feedbackSurveyData?.productIdentifier {
```

Comment:
Is it always the case the the promotionalOffer path comes from a feedback survey? Just to be sure

Reply:
Yes, you first need to explain the reason